### ISSUE TITLE
How to run as global process?

### ISSUE THREAD

**Issue by Gr1N** [SchedEx/SchedEx#13]
Hi,

Can you please advice on how to run SchedEx as a global process (something similar to [Quantum](https://hexdocs.pm/quantum/configuration.html#run-quantum-as-a-global-process)) in a cluster of nodes to prevent running same jobs multiple times on multiple nodes?

Thanks.

**Comment by mtrudel** [comment]
tl;dr: That's not something that SchedEx is designed to do.

Job supervision & mutexing is squarely outside of the problem that SchedEx sets out to solve. SchedEx's philosophy is to idiomatically (some might say dogmatically) follow OTP conventions regarding process semantics, and to allow the developer to compose SchedEx with whatever process management pattern they see fit (most typically using OTP {Dynamic,}Supervisors, but anything that manages `<PID>`s can work with SchedEx processes just fine; they're just self-contained GenServers under the hood). 

Where you might think of Quantum as a full-featured wood shop, SchedEx is more like a handheld circular saw. It's strictly less capable, but it's also possible to easily bring it into an existing application & solve the narrow problem of scheduling in a very clean & idiomatic way. Quantum is a great library that just happened to be much larger in scope than we originally needed, which is where SchedEx was born. If you need things like global processes, it sounds like that larger scope might be exactly what you need. 

If you *were* interested in rolling your own multi-node mutex on top of SchedEx, I'd suggest looking at Erlang's [global](http://erlang.org/doc/man/global.html) module, or (depending on the strictness of your needs), Elixir's [Registry](https://hexdocs.pm/elixir/Registry.html). FWIW, my personal hot take on this is that the problem of scheduling is pretty orthogonal to the problem of multi-node mutexing, and without a compelling reason to intertwine the two concerns I don't see a reason to solve them in the same library. OTP process semantics make a separation between them fairly straightforward, and would result in libraries which were more generally applicable and easier to reason about separately. 

Hope this helps! I'll be closing this issue since there's nothing to be done in SchedEx as a result of it. If subsequently you do end up having further issues or questions regarding SchedEx, please by all means do open another issue.
